tools/webpack-core/src/plugins/IModeljsLibraryExportsPlugin.ts

```diff
@@ -28,16 +28,22 @@ export class IModeljsLibraryExportsPlugin {
     });
     compiler.hooks.compilation.tap("IModeljsLibraryExportsPlugin", (compilation) => {
       compilation.moduleTemplates.javascript.hooks.content.tap("IModeljsLibraryExportsPlugin", (source, module) => {
-        if (!module.___IS_BENTLEY)
+        let currentModule = module;
+        // check for a rootModule (for concatenated and/or lazily loaded modules)
+        if (!currentModule.___IS_BENTLEY && currentModule.rootModule && currentModule.rootModule.___IS_BENTLEY) {
+            currentModule.rootModule.id = currentModule.id;
```

Comment:
Changing a module's `id` here seems like it could really break things in webpack...

Is there a way to accomplish the same thing without mutating any of these webpack objects?

Reply:
the issue is that the rootModule does not have an id....I suppose we could generate one instead. Still unsure how that might affect things as well. The thought is that this will always use the rootModule in the globals

Reply:
@wgoehrig, updated to avoid mutation. 

as an FYI, this is the plugin that is causing issues: https://webpack.js.org/plugins/module-concatenation-plugin/. The reason we have not seen this locally is that the plugin is disabled in dev and enabled in prod by default. We could disable it, but it could negatively affect performance. Without disabling it, theoretically any module that fits into the algorithm can be concatenated, which is why we need this check. We could just specifically check for all concatenated modules, but this limits to only modules that were flagged as modules that we want to hoist, since those are all that we care about.

Reply:
Yeah @kckst8, I agree - the potential performance benefits from scope hoisting outweigh the costs of having to work around it here.

LGTM